docs/development-environment.md

```diff
@@ -323,39 +323,57 @@ To get started, there are several ways to run the unit tests, depending on how y
 
 ## JavaScript unit tests
 
-Jetpack includes also several [Jest](https://jestjs.io/) based unit tests.
-To execute them in your local environment, you can use the following commands.
+The `jetpack test` command can be run from the monorepo's root to run a specific project's tests.
 
-* ### Admin Page unit tests
+This may be of limited benefit locally during development since it isn't possible to use watch mode or run tests only for an individual file.
 
-	Standing on your jetpack directory, run
+Each project within the monorepo may also have its own test commands, so an alternative is to `cd` into the project's root, and run the test commands from there.
 
-	```sh
-	pnpm install
-	pnpm test-client
-	pnpm test-gui
-	```
+### Packages
 
-* ### Jetpack modules unit tests
+Packages may have a package.json in the root that has a `scripts` entry, and this details the different types of test commands that can be run, `pnpm run test` is the usual command for JavaScript unit tests.
 
-	Standing on your jetpack directory, run
+For example, to run an individual test file in watch mode:
+```sh
+cd projects/packages/forms
+pnpm run test --watch -- path/to/test/file.js
+```
 
-	```sh
-	pnpm install
-	pnpm test-modules
-	```
+### Jetpack Plugin
 
-	You can also only run tests matching a specific pattern. To do that, use the argument `-g, --grep <pattern>`:
+The Jetpack plugin project also has some additional test commands that can be run from its root.
 
-	```sh
-	pnpm test-gui -g 'my custom pattern to filter tests'
-	```
+### Admin Page unit tests
 
-	To use a custom reporter, pass the argument `-R, --reporter <name>`:
+TODO - explain the purpose of these tests.
```

Comment:
It'd be good to add a brief explainer about these tests, but I don't personally know what they're for. The `gui` naming might mean visual regression or component testing, but I'm not sure.

Reply:
Looks like `test-client` tests stuff in `_inc/client/state/`, `_inc/client/lib/`, and `modules/`, while `test-gui` does `_inc/client/test/main.js` and `_inc/client/**/test/component.js`.

At a higher level, my impression is that these are both for the Jetpack dashboard and settings pages. `test-gui` is React components while `test-client` is modules and redux stuff. But someone from Vulcan might have a better overview.